docs/db/db-relations.md

```diff
@@ -0,0 +1,184 @@
+# Database Relations
+
+## Entity Relationship Diagram ERD
+
+An entity relationship diagram (ERD) shows the relationships of entity and sets stored in a database.
+
+An entity in this context is a table, this table contains a set of attributes, they can be of any data structure (INT, TEXT, ...).
+
+An entity can have several relations to other entities. Those relations are represented with "crow foots"
+
+### Relationships cardinality
+
+Relationships have a cardinality, normally they have two indicators that are shown on both sides of the line.
+
+1. One end of the line, refers to the **max** number of times that an instance of one entity can be associated with instances in the related entity. It can be **one**, **many** or **none**
+
+2. The second, describes the **min** number of times one instance can be related to others.
+It can be **zero** or **mandatory**
+
+The combination of those two indicators create relations, you can get more information [here](https://community.mis.temple.edu/mis3506digitaldesignfall2018/files/2018/10/Adam-Alalouf_Cardinality.pdf)
+
+![alt text](../img/db/db-relations.jpg "hierarchical roles")
+
+## Relations
+
+### Events
+
+Events will be the accumulation of the following tables :
+
+- `event`
+- `registration`
+- `registration_user`
+- `event_user`
+
+This part of the database takes care of schools activities with time.
+
+The `event` table contains the following relations:
+
+- `parentId` that is associated to himself. This association can be better explained if we take a look at the `path` column. An event can be based on a parent event (ex: the event `/madere/div-01/piscine-rust` is associated to this parent event `/madere/div-01`).
+
+- `registrationId` that is associated to the `registration` table. An event can have multiple registrations, but a registration can only be associated to one event, this being the parent event of the registration table.
```

Comment:
mmm this should be one to one I think - an event shouldn't have multiple registration, because the registration is responsible of the creation of the event (in the event service). For each registration, an event would be created (that's how the system works right now)

Reply:
oky, so this also means that an event **must** have a registration (the `registrationId` cannot be `null`)?

Reply:
yes it should always have a registration. You can create an event manually in db but it's not supposed to be done like that